packages/next/src/server/app-render/app-render.tsx

```diff
@@ -959,6 +961,7 @@ async function renderToHTMLOrFlightImpl(
         {
           onError: serverComponentsErrorHandler,
           nonce,
+          temporaryReferences: temporaryReferences,
```

Comment:
For SSR we shouldn't need temporary references since they'll be empty. I think you should omit this option here